src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,125 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.UserInvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        User sameUser = new User();
+        sameUser.setLogin("same_login");
+        sameUser.setAge(20);
+        sameUser.setPassword("same_password1234");
+        Storage.PEOPLE.add(sameUser);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.PEOPLE.remove(user);
+    }
+
+    @Test
+    void duplicatesLogin_NotOk() {
+        user = new User();
+        user.setPassword("joker2004");
+        user.setLogin("same_login");
+        user.setAge(28);
+        assertThrows(core.basesyntax.exceptions.UserInvalidDataException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void shortLogin_NotOk() {
+        user = new User();
+        user.setPassword("PASSWORD");
+        user.setLogin("LOG");
+        user.setAge(20);
+        assertThrows(core.basesyntax.exceptions.UserInvalidDataException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void shortPassword_NotOk() {
+        user = new User();
+        user.setPassword("spell");
+        user.setLogin("wizzard89");
+        user.setAge(25);
+        assertThrows(core.basesyntax.exceptions.UserInvalidDataException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void youngAge_NotOk() {
+        user = new User();
+        user.setPassword("Avadacedabra");
+        user.setLogin("Harry_Potter");
+        user.setAge(15);

```

Comment:
```suggestion
        user.setAge(17);
```
it is best practice to use thresholds for testing. age 18/17, login 6/5